src/k8s/pkg/snap/pebble.go

```diff
@@ -52,6 +55,30 @@ func (s *pebble) RestartService(ctx context.Context, name string) error {
 	return s.runCommand(ctx, []string{filepath.Join(s.snapDir, "bin", "pebble"), "restart", name})
 }
 
+func (s *pebble) Refresh(ctx context.Context, to types.RefreshOpts) error {
+	switch {
+	case to.Revision != "":
+		return fmt.Errorf("pebble does not support refreshing to a revision, only a local path")
+	case to.Channel != "":
+		return fmt.Errorf("pebble does not support refreshing to a channel, only a local path")
+	case to.LocalPath != "":
+		// replace the "kubernetes" binary with the new source.
```

Comment:
Huh, `LocalPath` will only upgrade the `kubernetes` binary?

What about all the other snap content?

Reply:
This is for pebble only and will be used for integration tests so we can check the kubernetes version change to verify. This gives us a smaller/more minimal image compared to having 2 copies of everything. We could implement this fully but pebble/image mode is not something we support officially.

Reply:
Hm, but `LocalPath` is somewhat part of our official API. I see people trying to refresh to a local build snap and wonder why this is not behaving as expected

Reply:
A local built snap will work just fine, see:
https://github.com/canonical/k8s-snap/blob/5f0c724546ab80ad53aaded7cb2b5400a61d1cb4/src/k8s/pkg/snap/snap.go#L84-L85

The pebble implementation is only used in CAPI testing with CAPD.

Reply:
My bad!